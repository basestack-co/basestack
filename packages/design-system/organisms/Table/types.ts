import { SpaceProps } from "styled-system";
import { PopupItems } from "../../molecules/Popup";

export type Col = {
  /**
   * Column title
   */
  title: string;
  /**
   * Optional if to be link
   */
  link?: string;
  /**
   * Optional image before title
   */
  image?: {
    src?: string;
    userName?: string;
  };
};

export type Row = {
  /**
   * PCols
   */
  cols: Array<Col>;
  /**
   * Popup items array
   */
  more: Array<PopupItems>;
};

export interface TableProps extends SpaceProps {
  /**
   * Table data
   */
  data: {
    headers: Array<string>;
    rows: Array<Row>;
  };
}

export interface RowProps {
  cols: Array<Col>;
  more: Array<PopupItems>;
  numberOfCols: number;
}
