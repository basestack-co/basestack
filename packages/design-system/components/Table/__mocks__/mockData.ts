export const mockTableData = {
  headers: ["Key", "Environment", "Description"],
  rows: [
    {
      cols: [
        {
          title: "key_dev_8299aa6b",
          link: "",
        },
        {
          title: "development",
          link: "",
        },
        {
          title: "development SDK",
          link: "",
        },
      ],
      more: [
        { text: "Edit", onClick: () => jest.fn },
        { text: "History", onClick: () => jest.fn },
        { text: "Delete", onClick: () => jest.fn },
      ],
    },
    {
      cols: [
        {
          title: "key_dev_8299aa6b",
          link: "",
        },
        {
          title: "development",
          link: "",
        },
        {
          title: "development SDK",
          link: "",
        },
      ],
      more: [
        { text: "Edit", onClick: () => jest.fn },
        { text: "History", onClick: () => jest.fn },
        { text: "Delete", onClick: () => jest.fn },
      ],
    },
  ],
};
