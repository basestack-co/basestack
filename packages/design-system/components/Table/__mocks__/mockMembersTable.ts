export const mockMore = [
  { text: "Edit", onClick: () => jest.fn },
  { text: "History", onClick: () => jest.fn },
  { text: "Delete", onClick: () => jest.fn },
];

export const mockMembersTable = {
  headers: ["Name", "Email", "Role"],
  rows: [
    {
      cols: [
        {
          image: {
            userName: "Jack",
            src: "",
          },
          title: "Jack",
        },
        {
          title: "jack@gmail.com",
        },
        {
          title: "Admin",
        },
      ],
      more: mockMore,
    },
    {
      cols: [
        {
          image: {
            userName: "Pedro",
            src: "",
          },
          title: "Pedro",
        },
        {
          title: "pedro_dev@gmail.com",
        },
        {
          title: "developer",
        },
      ],
      more: mockMore,
    },
    {
      cols: [
        {
          image: {
            userName: "Daniel",
            src: "",
          },
          title: "Daniel",
        },
        {
          title: "daniel_ninja_dev@gmail.com",
        },
        {
          title: "developer",
        },
      ],
      more: mockMore,
    },
    {
      cols: [
        {
          image: {
            userName: "Joana",
            src: "",
          },
          title: "Joana",
        },
        {
          title: "joana_mac@gmail.com",
        },
        {
          title: "QA",
        },
      ],
      more: mockMore,
    },
  ],
};
