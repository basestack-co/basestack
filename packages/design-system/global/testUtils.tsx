import { type RenderOptions, render } from "@testing-library/react";
import type React from "react";
// Styles
import { ThemeProvider } from "styled-components";
import theme from "../theme/lightTheme";

interface Props {
  children: React.ReactNode;
}

const WithThemeProvider = ({ children }: Props) => (
  <ThemeProvider theme={theme}>{children}</ThemeProvider>
);

const renderWithTheme = (
  ui: React.ReactElement,
  options?: Omit<RenderOptions, "wrapper">,
) => render(ui, { wrapper: WithThemeProvider, ...options });

export { renderWithTheme };
