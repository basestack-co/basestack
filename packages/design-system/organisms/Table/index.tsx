import React, { useCallback, useState } from "react";
import { useFloating, autoUpdate } from "@floating-ui/react-dom";
import { useTransition, animated, config } from "react-spring";
import get from "lodash.get";
import { Text, IconButton, Avatar } from "../../atoms";
import { scaleInTopRight } from "../../animations/springs";
import {
  Col,
  Container,
  ContentRow,
  Placeholder,
  StyledLink,
  StyledRow,
} from "./styles";
import { Popup } from "../../molecules";
import { TableProps, RowProps } from "./types";
import { useTheme } from "styled-components";

const AnimatedPopup = animated(Popup);

const Row = ({ cols = [], more, numberOfCols }: RowProps) => {
  const theme = useTheme();

  const [isPopupOpen, setIsPopupOpen] = useState(false);
  const { x, y, reference, floating, strategy } = useFloating({
    placement: "bottom-end",
    whileElementsMounted: autoUpdate,
  });

  const onClickMore = useCallback(() => {
    setIsPopupOpen((prevState) => !prevState);
  }, []);

  const transitionPopup = useTransition(isPopupOpen, {
    config: { ...config.default, duration: 150 },
    ...scaleInTopRight,
  });

  return (
    <StyledRow numberOfColumns={numberOfCols} data-testid="row">
      {cols &&
        cols.map((col, index) => {
          const imageSrc = get(col, "image.src");
          const imageUsername = get(col, "image.userName");
          return (
            <Col key={`${index.toString()}-col`}>
              {!!col.link ? (
                <StyledLink>
                  <Text
                    color={theme.colors.primary}
                    fontWeight="400"
                    size="small"
                  >
                    {col.title}
                  </Text>
                </StyledLink>
              ) : (
                <ContentRow>
                  {(!!imageSrc || !!imageUsername) && (
                    <Avatar
                      src={imageSrc}
                      userName={imageUsername}
                      alt={`${imageUsername} profile image`}
                      size="small"
                      mr={theme.spacing.s3}
                    />
                  )}
                  <Text fontWeight="400" size="small">
                    {col.title}
                  </Text>
                </ContentRow>
              )}
            </Col>
          );
        })}
      <Col>
        <IconButton ref={reference} icon="more_horiz" onClick={onClickMore} />
        {transitionPopup(
          (styles, item) =>
            item && (
              <AnimatedPopup
                style={styles}
                ref={floating}
                position={strategy}
                top={y}
                left={x}
                items={more}
              />
            )
        )}
      </Col>
    </StyledRow>
  );
};

const Table = ({ data, ...props }: TableProps) => {
  const numberOfCols = data.headers.length;
  return (
    <Container data-testid="table" {...props}>
      <StyledRow numberOfColumns={numberOfCols} data-testid="header">
        {data.headers &&
          data.headers.map((header, index) => {
            return (
              <Col key={`${index.toString()}-header`}>
                <Text muted size="small">
                  {header}
                </Text>
              </Col>
            );
          })}
        <Placeholder />
      </StyledRow>
      {data.rows &&
        data.rows.map((row, index) => {
          return (
            <Row
              key={`${index.toString()}-row`}
              cols={row.cols}
              more={row.more}
              numberOfCols={numberOfCols}
            />
          );
        })}
    </Container>
  );
};

export default Table;
