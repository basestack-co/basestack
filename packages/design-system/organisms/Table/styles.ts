import styled from "styled-components";
import { space } from "styled-system";
import { rem } from "polished";

export const Container = styled.div`
  ${space};
  background-color: ${({ theme }) => theme.colors.gray50};
  padding: ${({ theme }) => theme.spacing.s5};
  border-radius: 4px;
`;

export const StyledRow = styled.div<{ numberOfColumns: number }>`
  display: grid;
  grid-template-columns:
    repeat(${({ numberOfColumns }) => numberOfColumns || 3}, 1fr)
    auto;
  grid-gap: ${({ theme }) => theme.spacing.s5};

  &:not(:last-child) {
    border-bottom: 1px solid ${({ theme }) => theme.colors.gray200};
  }
`;

export const Placeholder = styled.div`
  min-width: ${rem("36px")};
  pointer-events: none;
`;

export const Col = styled.div`
  height: ${rem("45px")};
  display: flex;
  flex-direction: column;
  justify-content: center;
`;

export const StyledLink = styled.a`
  text-decoration: none;
  cursor: pointer;

  &:hover {
    text-decoration: underline;
    color: ${({ theme }) => theme.colors.primary};
  }
`;

export const ContentRow = styled.div`
  display: flex;
  align-items: center;
`;
