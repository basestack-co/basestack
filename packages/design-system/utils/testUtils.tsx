import defaultTheme from "@basestack/design-system/theme/lightTheme";
import isPropValid from "@emotion/is-prop-valid";
import { RenderResult, render } from "@testing-library/react";
import React from "react";
import { StyleSheetManager, ThemeProvider } from "styled-components";

const Provider = ({
  children,
  theme = defaultTheme,
}: {
  children: React.ReactNode;
  theme: any;
}) => (
  <ThemeProvider theme={theme}>
    <StyleSheetManager shouldForwardProp={isPropValid}>
      {children}
    </StyleSheetManager>
  </ThemeProvider>
);

export const renderWithTheme = (
  ui: React.ReactElement,
  options?: any,
): RenderResult => render(ui, { wrapper: Provider, ...options });
