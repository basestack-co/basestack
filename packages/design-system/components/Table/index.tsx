import React from "react";
import { useTheme } from "styled-components";
import { animated } from "react-spring";
import { getValue } from "@basestack/utils";
import { useFloatingPopup } from "@basestack/hooks";
// Components
import Text from "../Text";
import IconButton from "../IconButton";
import Avatar from "../Avatar";
import Popup from "../Popup";
import {
  Col,
  Container,
  ContentRow,
  PopupWrapper,
  Placeholder,
  StyledLink,
  StyledRow,
} from "./styles";
import { TableProps, RowProps } from "./types";

const AnimatedPopup = animated(Popup);

const Row = ({ cols = [], more, numberOfCols }: RowProps) => {
  const theme = useTheme();

  const {
    popupWrapperRef,
    x,
    y,
    refs,
    strategy,
    transition,
    getReferenceProps,
    getFloatingProps,
    onClickMore,
    onCloseMenu,
  } = useFloatingPopup();

  return (
    <StyledRow numberOfColumns={numberOfCols} data-testid="row">
      {cols &&
        cols.map((col, index) => {
          const imageSrc = getValue(col, "image.src");
          const imageUsername = getValue(col, "image.userName");
          return (
            <Col key={`${index.toString()}-col`}>
              {!!col.link ? (
                <StyledLink>
                  <Text
                    color={theme.colors.primary}
                    fontWeight="400"
                    size="small"
                  >
                    {col.title}
                  </Text>
                </StyledLink>
              ) : (
                <ContentRow>
                  {(!!imageSrc || !!imageUsername) && (
                    <Avatar
                      src={imageSrc}
                      userName={imageUsername}
                      alt={`${imageUsername} profile image`}
                      size="small"
                      mr={theme.spacing.s3}
                    />
                  )}
                  <Text fontWeight="400" size="small">
                    {col.title}
                  </Text>
                </ContentRow>
              )}
            </Col>
          );
        })}
      <Col>
        {more.length > 0 ? (
          <PopupWrapper ref={popupWrapperRef}>
            <IconButton
              {...getReferenceProps}
              ref={refs.setReference}
              icon="more_horiz"
              onClick={onClickMore}
            />
            {transition(
              (styles, item) =>
                item && (
                  <AnimatedPopup
                    {...getFloatingProps}
                    ref={refs.setFloating}
                    style={styles}
                    position={strategy}
                    top={y}
                    left={x}
                    items={more}
                    onClickList={onCloseMenu}
                  />
                ),
            )}
          </PopupWrapper>
        ) : (
          <Placeholder />
        )}
      </Col>
    </StyledRow>
  );
};

const Table = ({ data, ...props }: TableProps) => {
  const numberOfCols = data.headers.length;
  return (
    <Container data-testid="table" {...props}>
      <StyledRow numberOfColumns={numberOfCols} data-testid="header">
        {data.headers &&
          data.headers.map((header, index) => {
            return (
              <Col key={`${index.toString()}-header`}>
                <Text muted size="small">
                  {header}
                </Text>
              </Col>
            );
          })}
        <Placeholder />
      </StyledRow>
      {data.rows &&
        data.rows.map((row, index) => {
          return (
            <Row
              key={`${index.toString()}-row`}
              cols={row.cols}
              more={data.rows.length > 1 ? row.more : []}
              numberOfCols={numberOfCols}
            />
          );
        })}
    </Container>
  );
};

export default Table;
