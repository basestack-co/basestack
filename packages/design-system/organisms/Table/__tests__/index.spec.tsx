import { cleanup } from "@testing-library/react";
import { renderWithTheme } from "../../../utils/testUtils";
import Table from "../index";
import { mockTableData } from "../__mocks__/mockData";
import { mockMembersTable } from "../__mocks__/mockMembersTable";

describe("Table Organism tests", () => {
  afterEach(cleanup);

  test("should render Table correctly", () => {
    const { asFragment } = renderWithTheme(<Table data={mockTableData} />);
    expect(asFragment()).toMatchSnapshot();
  });

  test("should render Table with space utils", () => {
    const { asFragment } = renderWithTheme(
      <Table data={mockTableData} mt={30} />,
    );
    expect(asFragment()).toMatchSnapshot();
  });

  test("should render Table with images", () => {
    const { asFragment, getAllByTestId } = renderWithTheme(
      <Table data={mockMembersTable} />,
    );

    const avatars = getAllByTestId("avatar");

    expect(avatars).toHaveLength(4);
    avatars.forEach((avatar) => expect(avatar).toBeVisible());
    expect(asFragment()).toMatchSnapshot();
  });
});
