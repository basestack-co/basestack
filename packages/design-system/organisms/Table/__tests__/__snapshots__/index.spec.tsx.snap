// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Table Organism tests should render Table correctly 1`] = `
<DocumentFragment>
  .c3 {
  font-size: 0.875rem;
  line-height: 1.375rem;
  font-weight: 400;
  font-family: 'Roboto',sans-serif;
  color: #757575;
}

.c6 {
  font-size: 0.875rem;
  line-height: 1.375rem;
  font-weight: 400;
  font-weight: 400;
  font-family: 'Roboto',sans-serif;
  color: #000000;
}

.c8 {
  height: 1.5rem;
  width: 1.5rem;
  -webkit-flex-shrink: 0;
  -ms-flex-negative: 0;
  flex-shrink: 0;
}

.c9 {
  color: #000000;
  font-size: 1.5rem;
}

.c7 {
  border: none;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  cursor: pointer;
  border-radius: 50%;
  -webkit-transition: background-color 0.1s ease-in-out,color 0.1s ease-in-out, border-color 0.1s ease-in-out;
  transition: background-color 0.1s ease-in-out,color 0.1s ease-in-out, border-color 0.1s ease-in-out;
  background-color: transparent;
  color: #000000;
  height: 2rem;
  width: 2rem;
}

.c7 .material-symbols-sharp {
  color: inherit;
}

.c7:hover {
  background-color: #EEEEEE;
}

.c7:active {
  background-color: #E2E2E2;
}

.c0 {
  background-color: #F6F6F6;
  padding: 1.25rem;
  border-radius: 4px;
}

.c1 {
  display: grid;
  grid-template-columns: repeat(3,1fr) auto;
  grid-gap: 1.25rem;
}

.c1:not(:last-child) {
  border-bottom: 1px solid #E2E2E2;
}

.c4 {
  min-width: 2.25rem;
  pointer-events: none;
}

.c2 {
  height: 2.8125rem;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
}

.c5 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

<div
    class="c0"
    data-testid="table"
  >
    <div
      class="c1"
      data-testid="header"
    >
      <div
        class="c2"
      >
        <p
          class="c3"
        >
          Key
        </p>
      </div>
      <div
        class="c2"
      >
        <p
          class="c3"
        >
          Environment
        </p>
      </div>
      <div
        class="c2"
      >
        <p
          class="c3"
        >
          Description
        </p>
      </div>
      <div
        class="c4"
      />
    </div>
    <div
      class="c1"
      data-testid="row"
    >
      <div
        class="c2"
      >
        <div
          class="c5"
        >
          <p
            class="c6"
            font-weight="400"
          >
            key_dev_8299aa6b
          </p>
        </div>
      </div>
      <div
        class="c2"
      >
        <div
          class="c5"
        >
          <p
            class="c6"
            font-weight="400"
          >
            development
          </p>
        </div>
      </div>
      <div
        class="c2"
      >
        <div
          class="c5"
        >
          <p
            class="c6"
            font-weight="400"
          >
            development SDK
          </p>
        </div>
      </div>
      <div
        class="c2"
      >
        <button
          class="c7"
        >
          <div
            class="c8"
            data-testid="icon-container"
            size="24"
          >
            <span
              class="c9 material-symbols-sharp"
              color="#000000"
              data-testid="icon"
              size="24"
            >
              more_horiz
            </span>
          </div>
        </button>
      </div>
    </div>
    <div
      class="c1"
      data-testid="row"
    >
      <div
        class="c2"
      >
        <div
          class="c5"
        >
          <p
            class="c6"
            font-weight="400"
          >
            key_dev_8299aa6b
          </p>
        </div>
      </div>
      <div
        class="c2"
      >
        <div
          class="c5"
        >
          <p
            class="c6"
            font-weight="400"
          >
            development
          </p>
        </div>
      </div>
      <div
        class="c2"
      >
        <div
          class="c5"
        >
          <p
            class="c6"
            font-weight="400"
          >
            development SDK
          </p>
        </div>
      </div>
      <div
        class="c2"
      >
        <button
          class="c7"
        >
          <div
            class="c8"
            data-testid="icon-container"
            size="24"
          >
            <span
              class="c9 material-symbols-sharp"
              color="#000000"
              data-testid="icon"
              size="24"
            >
              more_horiz
            </span>
          </div>
        </button>
      </div>
    </div>
  </div>
</DocumentFragment>
`;

exports[`Table Organism tests should render Table with images 1`] = `
<DocumentFragment>
  .c6 {
  height: 1.75rem;
  width: 1.75rem;
  border-radius: 50%;
  overflow: hidden;
  -webkit-flex-shrink: 0;
  -ms-flex-negative: 0;
  flex-shrink: 0;
  margin-right: 0.75rem;
  border-radius: 50%;
}

.c7 {
  height: 100%;
  width: 100%;
  object-fit: cover;
}

.c3 {
  font-size: 0.875rem;
  line-height: 1.375rem;
  font-weight: 400;
  font-family: 'Roboto',sans-serif;
  color: #757575;
}

.c8 {
  font-size: 0.875rem;
  line-height: 1.375rem;
  font-weight: 400;
  font-weight: 400;
  font-family: 'Roboto',sans-serif;
  color: #000000;
}

.c10 {
  height: 1.5rem;
  width: 1.5rem;
  -webkit-flex-shrink: 0;
  -ms-flex-negative: 0;
  flex-shrink: 0;
}

.c11 {
  color: #000000;
  font-size: 1.5rem;
}

.c9 {
  border: none;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  cursor: pointer;
  border-radius: 50%;
  -webkit-transition: background-color 0.1s ease-in-out,color 0.1s ease-in-out, border-color 0.1s ease-in-out;
  transition: background-color 0.1s ease-in-out,color 0.1s ease-in-out, border-color 0.1s ease-in-out;
  background-color: transparent;
  color: #000000;
  height: 2rem;
  width: 2rem;
}

.c9 .material-symbols-sharp {
  color: inherit;
}

.c9:hover {
  background-color: #EEEEEE;
}

.c9:active {
  background-color: #E2E2E2;
}

.c0 {
  background-color: #F6F6F6;
  padding: 1.25rem;
  border-radius: 4px;
}

.c1 {
  display: grid;
  grid-template-columns: repeat(3,1fr) auto;
  grid-gap: 1.25rem;
}

.c1:not(:last-child) {
  border-bottom: 1px solid #E2E2E2;
}

.c4 {
  min-width: 2.25rem;
  pointer-events: none;
}

.c2 {
  height: 2.8125rem;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
}

.c5 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

<div
    class="c0"
    data-testid="table"
  >
    <div
      class="c1"
      data-testid="header"
    >
      <div
        class="c2"
      >
        <p
          class="c3"
        >
          Name
        </p>
      </div>
      <div
        class="c2"
      >
        <p
          class="c3"
        >
          Email
        </p>
      </div>
      <div
        class="c2"
      >
        <p
          class="c3"
        >
          Role
        </p>
      </div>
      <div
        class="c4"
      />
    </div>
    <div
      class="c1"
      data-testid="row"
    >
      <div
        class="c2"
      >
        <div
          class="c5"
        >
          <div
            class="c6"
            data-testid="avatar"
            size="28"
          >
            <img
              alt="Jack profile image"
              class="c7"
              src="https://avatars.dicebear.com/api/initials/Jack.svg?size=28&fontSize=40"
            />
          </div>
          <p
            class="c8"
            font-weight="400"
          >
            Jack
          </p>
        </div>
      </div>
      <div
        class="c2"
      >
        <div
          class="c5"
        >
          <p
            class="c8"
            font-weight="400"
          >
            jack@gmail.com
          </p>
        </div>
      </div>
      <div
        class="c2"
      >
        <div
          class="c5"
        >
          <p
            class="c8"
            font-weight="400"
          >
            Admin
          </p>
        </div>
      </div>
      <div
        class="c2"
      >
        <button
          class="c9"
        >
          <div
            class="c10"
            data-testid="icon-container"
            size="24"
          >
            <span
              class="c11 material-symbols-sharp"
              color="#000000"
              data-testid="icon"
              size="24"
            >
              more_horiz
            </span>
          </div>
        </button>
      </div>
    </div>
    <div
      class="c1"
      data-testid="row"
    >
      <div
        class="c2"
      >
        <div
          class="c5"
        >
          <div
            class="c6"
            data-testid="avatar"
            size="28"
          >
            <img
              alt="Pedro profile image"
              class="c7"
              src="https://avatars.dicebear.com/api/initials/Pedro.svg?size=28&fontSize=40"
            />
          </div>
          <p
            class="c8"
            font-weight="400"
          >
            Pedro
          </p>
        </div>
      </div>
      <div
        class="c2"
      >
        <div
          class="c5"
        >
          <p
            class="c8"
            font-weight="400"
          >
            pedro_dev@gmail.com
          </p>
        </div>
      </div>
      <div
        class="c2"
      >
        <div
          class="c5"
        >
          <p
            class="c8"
            font-weight="400"
          >
            developer
          </p>
        </div>
      </div>
      <div
        class="c2"
      >
        <button
          class="c9"
        >
          <div
            class="c10"
            data-testid="icon-container"
            size="24"
          >
            <span
              class="c11 material-symbols-sharp"
              color="#000000"
              data-testid="icon"
              size="24"
            >
              more_horiz
            </span>
          </div>
        </button>
      </div>
    </div>
    <div
      class="c1"
      data-testid="row"
    >
      <div
        class="c2"
      >
        <div
          class="c5"
        >
          <div
            class="c6"
            data-testid="avatar"
            size="28"
          >
            <img
              alt="Daniel profile image"
              class="c7"
              src="https://avatars.dicebear.com/api/initials/Daniel.svg?size=28&fontSize=40"
            />
          </div>
          <p
            class="c8"
            font-weight="400"
          >
            Daniel
          </p>
        </div>
      </div>
      <div
        class="c2"
      >
        <div
          class="c5"
        >
          <p
            class="c8"
            font-weight="400"
          >
            daniel_ninja_dev@gmail.com
          </p>
        </div>
      </div>
      <div
        class="c2"
      >
        <div
          class="c5"
        >
          <p
            class="c8"
            font-weight="400"
          >
            developer
          </p>
        </div>
      </div>
      <div
        class="c2"
      >
        <button
          class="c9"
        >
          <div
            class="c10"
            data-testid="icon-container"
            size="24"
          >
            <span
              class="c11 material-symbols-sharp"
              color="#000000"
              data-testid="icon"
              size="24"
            >
              more_horiz
            </span>
          </div>
        </button>
      </div>
    </div>
    <div
      class="c1"
      data-testid="row"
    >
      <div
        class="c2"
      >
        <div
          class="c5"
        >
          <div
            class="c6"
            data-testid="avatar"
            size="28"
          >
            <img
              alt="Joana profile image"
              class="c7"
              src="https://avatars.dicebear.com/api/initials/Joana.svg?size=28&fontSize=40"
            />
          </div>
          <p
            class="c8"
            font-weight="400"
          >
            Joana
          </p>
        </div>
      </div>
      <div
        class="c2"
      >
        <div
          class="c5"
        >
          <p
            class="c8"
            font-weight="400"
          >
            joana_mac@gmail.com
          </p>
        </div>
      </div>
      <div
        class="c2"
      >
        <div
          class="c5"
        >
          <p
            class="c8"
            font-weight="400"
          >
            QA
          </p>
        </div>
      </div>
      <div
        class="c2"
      >
        <button
          class="c9"
        >
          <div
            class="c10"
            data-testid="icon-container"
            size="24"
          >
            <span
              class="c11 material-symbols-sharp"
              color="#000000"
              data-testid="icon"
              size="24"
            >
              more_horiz
            </span>
          </div>
        </button>
      </div>
    </div>
  </div>
</DocumentFragment>
`;

exports[`Table Organism tests should render Table with space utils 1`] = `
<DocumentFragment>
  .c3 {
  font-size: 0.875rem;
  line-height: 1.375rem;
  font-weight: 400;
  font-family: 'Roboto',sans-serif;
  color: #757575;
}

.c6 {
  font-size: 0.875rem;
  line-height: 1.375rem;
  font-weight: 400;
  font-weight: 400;
  font-family: 'Roboto',sans-serif;
  color: #000000;
}

.c8 {
  height: 1.5rem;
  width: 1.5rem;
  -webkit-flex-shrink: 0;
  -ms-flex-negative: 0;
  flex-shrink: 0;
}

.c9 {
  color: #000000;
  font-size: 1.5rem;
}

.c7 {
  border: none;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  cursor: pointer;
  border-radius: 50%;
  -webkit-transition: background-color 0.1s ease-in-out,color 0.1s ease-in-out, border-color 0.1s ease-in-out;
  transition: background-color 0.1s ease-in-out,color 0.1s ease-in-out, border-color 0.1s ease-in-out;
  background-color: transparent;
  color: #000000;
  height: 2rem;
  width: 2rem;
}

.c7 .material-symbols-sharp {
  color: inherit;
}

.c7:hover {
  background-color: #EEEEEE;
}

.c7:active {
  background-color: #E2E2E2;
}

.c0 {
  margin-top: 30px;
  background-color: #F6F6F6;
  padding: 1.25rem;
  border-radius: 4px;
}

.c1 {
  display: grid;
  grid-template-columns: repeat(3,1fr) auto;
  grid-gap: 1.25rem;
}

.c1:not(:last-child) {
  border-bottom: 1px solid #E2E2E2;
}

.c4 {
  min-width: 2.25rem;
  pointer-events: none;
}

.c2 {
  height: 2.8125rem;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
}

.c5 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

<div
    class="c0"
    data-testid="table"
  >
    <div
      class="c1"
      data-testid="header"
    >
      <div
        class="c2"
      >
        <p
          class="c3"
        >
          Key
        </p>
      </div>
      <div
        class="c2"
      >
        <p
          class="c3"
        >
          Environment
        </p>
      </div>
      <div
        class="c2"
      >
        <p
          class="c3"
        >
          Description
        </p>
      </div>
      <div
        class="c4"
      />
    </div>
    <div
      class="c1"
      data-testid="row"
    >
      <div
        class="c2"
      >
        <div
          class="c5"
        >
          <p
            class="c6"
            font-weight="400"
          >
            key_dev_8299aa6b
          </p>
        </div>
      </div>
      <div
        class="c2"
      >
        <div
          class="c5"
        >
          <p
            class="c6"
            font-weight="400"
          >
            development
          </p>
        </div>
      </div>
      <div
        class="c2"
      >
        <div
          class="c5"
        >
          <p
            class="c6"
            font-weight="400"
          >
            development SDK
          </p>
        </div>
      </div>
      <div
        class="c2"
      >
        <button
          class="c7"
        >
          <div
            class="c8"
            data-testid="icon-container"
            size="24"
          >
            <span
              class="c9 material-symbols-sharp"
              color="#000000"
              data-testid="icon"
              size="24"
            >
              more_horiz
            </span>
          </div>
        </button>
      </div>
    </div>
    <div
      class="c1"
      data-testid="row"
    >
      <div
        class="c2"
      >
        <div
          class="c5"
        >
          <p
            class="c6"
            font-weight="400"
          >
            key_dev_8299aa6b
          </p>
        </div>
      </div>
      <div
        class="c2"
      >
        <div
          class="c5"
        >
          <p
            class="c6"
            font-weight="400"
          >
            development
          </p>
        </div>
      </div>
      <div
        class="c2"
      >
        <div
          class="c5"
        >
          <p
            class="c6"
            font-weight="400"
          >
            development SDK
          </p>
        </div>
      </div>
      <div
        class="c2"
      >
        <button
          class="c7"
        >
          <div
            class="c8"
            data-testid="icon-container"
            size="24"
          >
            <span
              class="c9 material-symbols-sharp"
              color="#000000"
              data-testid="icon"
              size="24"
            >
              more_horiz
            </span>
          </div>
        </button>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
